### PR TITLE
Fixed and updated links throughout the documentation

### PR DESCRIPTION
**Pull request by evildmp**
### Summary

Numerous links in the documentation were outdated or broken; this improves them.

[ci only docs]

### CODE REVIEW

File: docs/contributing/code.rst [hunk 1]
```diff
@@ -134,7 +131,7 @@ django CMS you need to have the following dependencies installed:
     1. `Node <https://nodejs.org/>`_ version 0.12.7 (will install npm as well).
        We recommend using `NVM <https://github.com/creationix/nvm>`_ to get
        the correct version of Node.
-    2. `Globally installed gulp <https://github.com/gulpjs/gulp/blob/master/docs/getting-started.md#1-install-gulp-globally>`_
+    2. `gulp <https://github.com/gulpjs/gulp/blob/master/docs/getting-started.md>`_
```
Comment (FinalAngel) [inline docs/contributing/code.rst]: gulp is written `Gulp`

File: docs/contributing/code.rst [hunk 2]
```diff
@@ -143,7 +140,7 @@ Styles
 We use `Sass <http://sass-lang.com/>`_ for our styles. The files
 are located within ``cms/static/cms/sass`` and can be compiled using the
 `libsass <http://libsass.org/>`_ implementation of Sass compiler through
-`Gulp <http://gulpjs.com/>`_.
+`gulp <http://gulpjs.com/>`_.
```
Comment (FinalAngel) [inline docs/contributing/code.rst]: same here

File: docs/contributing/code.rst (hunk 1 above)
Comment (evildmp) [inline docs/contributing/code.rst]: On https://github.com/gulpjs/gulp/blob/master/docs/API.md and http://gulpjs.com ("gulp is a toolkit") it's mostly not.

Comment (FinalAngel) [inline docs/contributing/code.rst]: I'm fine with either, the "command" is "gulp" the logo is "Gulp" (http://gulpjs.com/)

**Pull request merged**